webview-ui/src/components/settings/ExperimentalSettings.tsx

```diff
@@ -74,6 +86,92 @@ export const ExperimentalSettings = ({
 								{t("settings:experimental.autoCondenseContextPercent.description")}
 							</div>
 						</div>
+
+						{/* New API Configuration Selection */}
+						<div className="flex flex-col gap-3">
+							<div className="flex items-center gap-4 font-bold">
+								<span className="codicon codicon-settings-gear" />
+								<div>{t("settings:experimental.condensingApiConfiguration.label")}</div>
+							</div>
+							<div>
+								<div className="text-[13px] text-vscode-descriptionForeground mb-2">
+									{t("settings:experimental.condensingApiConfiguration.description")}
+								</div>
+								<Select
+									value={condensingApiConfigId || "-"}
+									onValueChange={(value) => {
+										const newConfigId = value === "-" ? "" : value
+										setCondensingApiConfigId(newConfigId)
+										vscode.postMessage({
+											type: "condensingApiConfigId",
+											text: newConfigId,
+										})
+									}}>
+									<SelectTrigger className="w-full">
+										<SelectValue
+											placeholder={t(
+												"settings:experimental.condensingApiConfiguration.useCurrentConfig",
+											)}
+										/>
+									</SelectTrigger>
+									<SelectContent>
+										<SelectItem value="-">
+											{t("settings:experimental.condensingApiConfiguration.useCurrentConfig")}
+										</SelectItem>
+										{(listApiConfigMeta || []).map((config) => (
+											<SelectItem key={config.id} value={config.id}>
+												{config.name}
+											</SelectItem>
+										))}
+									</SelectContent>
+								</Select>
+							</div>
+						</div>
+
+						{/* Custom Prompt Section */}
+						<div className="flex flex-col gap-3">
+							<div className="flex items-center gap-4 font-bold">
+								<span className="codicon codicon-edit" />
+								<div>{t("settings:experimental.customCondensingPrompt.label")}</div>
+							</div>
+							<div>
+								<div className="text-[13px] text-vscode-descriptionForeground mb-2">
+									{t("settings:experimental.customCondensingPrompt.description")}
+								</div>
+								<VSCodeTextArea
+									resize="vertical"
+									value={customCondensingPrompt || ""}
```

Comment:
I think this would be easier to use if the value defaulted to `SUMMARY_PROMPT` to make it obvious what the default is and to make it easier for people to make minor adjustments. In this case we wouldn't need a placeholder, and hitting the reset button would just populate with the default value.